modules/images/regenerate-existing-images/load.php

```diff
@@ -7,3 +7,79 @@
  * @since n.e.x.t
  * @package performance-lab
  */
+
+/**
+ * Register the background job taxonomy.
+ *
+ * Intentionally on lower priority, so that other post types can be
+ * registered successfully and taxonomy can be used for all the post types.
+ *
+ * This taxonomy will be used as a queue system where each term represents a job in the queue.
+ *
+ * @return void
+ */
+function perflab_background_job_taxonomy() {
+	// Get all registered public post types.
+	$object_types = get_post_types(
+		array(
+			'public' => true,
+		),
+		'names'
+	);
+
+	// Labels for the background job taxonomy.
+	$labels = array(
+		'name'              => _x( 'Background Jobs', 'taxonomy general name', 'performance-lab' ),
+		'singular_name'     => _x( 'Background Job', 'taxonomy singular name', 'performance-lab' ),
+		'search_items'      => __( 'Search Background Jobs', 'performance-lab' ),
+		'all_items'         => __( 'All Background Jobs', 'performance-lab' ),
+		'parent_item'       => __( 'Parent Background Job', 'performance-lab' ),
+		'parent_item_colon' => __( 'Parent Background Job:', 'performance-lab' ),
+		'edit_item'         => __( 'Edit Background Job', 'performance-lab' ),
+		'update_item'       => __( 'Update Background Job', 'performance-lab' ),
+		'add_new_item'      => __( 'Add New Background Job', 'performance-lab' ),
+		'new_item_name'     => __( 'New Background Job Name', 'performance-lab' ),
+		'menu_name'         => __( 'Background Job', 'performance-lab' ),
+	);
+
+	$caps = get_background_job_capabilities();
+
+	// Taxonomy arguments.
+	$args = array(
+		'hierarchical'       => false, // We do not need child job.
+		'labels'             => $labels,
+		'public'             => false,
+		'show_ui'            => true,
+		'show_in_menu'       => false,
+		'show_in_nav_menus'  => true,
+		'show_in_quick_edit' => false,
+		'show_admin_column'  => false,
+		'show_in_rest'       => false,
+		'query_var'          => true,
+		'rewrite'            => array( 'slug' => 'background_job' ),
+		'capabilities'       => array(
+			'manage_terms' => 'manage_options',
+			'edit_terms'   => $caps['edit'],
+			'delete_terms' => $caps['delete'],
+			'assign_terms' => $caps['assign'],
+		),
```

Comment:
It's a bit over-complicated to manually assign this here for each field. Why not change the capabilities function to simply return the data with the right keys? That way you can simply change it to something like
```suggestion
		'capabilities'       => perflab_get_background_job_capabilities(),
```